lib/web_ui/lib/src/engine/pointer_binding.dart

```diff
@@ -417,24 +443,25 @@ class PointerAdapter extends BaseAdapter {
 }
 
 /// Adapter to be used with browsers that support touch events.
-class TouchAdapter extends BaseAdapter {
-  TouchAdapter(
-    PointerDataCallback callback,
-    DomRenderer domRenderer,
+class _TouchAdapter extends _BaseAdapter {
+  _TouchAdapter(
+    _PointerDataCallback callback,
+    html.Element glassPaneElement,
     PointerDataConverter _pointerDataConverter
-  ) : super(callback, domRenderer, _pointerDataConverter);
+  ) : super(callback, glassPaneElement, _pointerDataConverter);
+
+  bool _pressed = false;
 
   @override
-  void _setup() {
+  void setup() {
     _addEventListener('touchstart', (html.Event event) {
-      _updateButtonDownState(
-          _deviceFromHtmlEvent(event), _kPrimaryMouseButton, true);
+      _pressed = true;
```

Comment:
Having a single boolean will likely not work. What happens when two fingers are touching the screen? Each finger will send its own `touchstart` event.

Reply:
Changed to a set of int `_pressedTouches`, and also added tests